components/Column.js

```diff
@@ -77,30 +77,56 @@ export default class Column extends Component {
   }
 
   // Renders the "bricks" within the columns
-  // _renderBricks :: [images] -> [TouchableTag || ImageTag...]
-  _renderBricks (bricks) {
-    return bricks.map((brick, index) => {
-      const gutter = (index === 0) ? 0 : brick.gutter;
-      const key = `RN-MASONRY-BRICK-${brick.column}-${index}`;
+  // _renderBrick :: images -> [TouchableTag || ImageTag...]
+  _renderBrick (data) {
+      // Example Data Structure
+      // {
+      //   "item": {
+      //     "uri": "https://img.buzzfeed.com/buzzfeed-static/static/2016-01/14/20/campaign_images/webdr15/which-delicious-mexican-food-item-are-you-based-o-2-20324-1452822970-1_dblbig.jpg",
+      //     "column": 0,
+      //     "dimensions": {
+      //       "width": 625,
+      //       "height": 415
+      //     },
+      //     "width": 180.675,
+      //     "height": 119.96820000000001,
+      //     "gutter": 3.65
+      //   },
+      //   "index": 9
+      // }
+      const brick = data.item;
+      const gutter = (data.index === 0) ? 0 : brick.gutter;
+      const key = `RN-MASONRY-BRICK-${brick.column}-${data.index}`;
       const props = { ...brick, gutter, key };
 
       return (
         <Brick
           {...props} />
       );
-    });
   }
+  
+  // _keyExtractor :: item -> id
+  _keyExtractor = (item) => (item.id || item.key);
 
   render() {
     return (
       <View
-        key={this.props.columnKey}
         style={[
-          { width: this.state.columnWidth },
+          { 
+            width: this.state.columnWidth,
+            overflow: 'hidden'
+          },
           styles.masonry__column
         ]}>
-          {this._renderBricks(this.state.images)}
+        <FlatList
+          key={this.props.columnKey}
+          data={this.state.images}
+          keyExtractor={this._keyExtractor}
+          renderItem={this._renderBrick}
+        />
       </View>
```

Comment:
I think I'm getting most of this but I'm still a little confused on this part. How does this render, for example when columns = 2, into 

```
|            Masonry (FlatList)        |
----------------------------------------
| Column(FlatList) |  Column(FlatList) |
|     ++++++++     |     ++++++++      | 
|     ++++++++     |     ++++++++      | 
|  -------------   |     ++++++++      | 
|     ++++++++     |     ++++++++      | 
|     ++++++++     |  -------------    |
|  -------------   |     ++++++++      |
|     ++++++++     |     ++++++++      | 
|     ++++++++     |  -------------    |
```

I only see 1 parent `View` with a, now, `FlatList` that will render multiple `Bricks` and the parent View is set to style `flexDirection: 'column'`. I would think this would only render 1 vertical column,...not 2. Thanks for you patience as I learn more RN.

Reply:
ListView and FlatList render a row for each item in the dataSource. So in `Masonry` component, if you recall resolveBricks as well as the logic you've implemented, we are essentially gathering all of the data and creating a nested array. Each nested array indicates a column, which is therefore rendered as a "row" within ListView (FlatList as well).

Reply:
```
const expectedData = [ <--Masonry renders 1 large row?
    [assignedBricks[0], assignedBricks[2]], //column 0 <--Column renders as FlastList
    [assignedBricks[1], assignedBricks[3]] //column 1 <--Column renders as FlastList
  ];
```
So in this example data, I follow the nested arrays and they represent columns. What I'm struggling with is I thought Lists renders rows of equal "max heights".  So how what causes the columns to be side by side?

Reply:
Does the new approach solve problems like this: https://github.com/brh55/react-native-masonry/issues/41 ?

Reply:
@srameshr That is intentional as it doesn't account for current height of all the elements -- this would require some work --, it just simply places the items into the columns based on the index of its original source.

Reply:
@kmcgill88 The columns width are divided between the parent element's width. Hence why they stay side-by-side. The `FlatList` renders the minimum height of all the child elements.

Reply:
@brh55 How could we fix this?

Reply:
@srameshr We create a reference to each columns height based on the accumulated image heights, and insert the images according to shortest column of that given time.

Reply:
@brh55 Will that break the sorted option?

Reply:
@srameshr Yeah so you need to have a way to define the arrangement, height, order, reverse, etc.